### PR TITLE
ARROW-12985: [Python][Packaging] Unable to install pygit2 in the arm64 wheel builds

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: dev/archery/archery/crossbow/core.py [hunk 1]
```diff
@@ -194,7 +194,8 @@ def credentials(self, url, username_from_url, allowed_types):
             print(msg)
             raise CrossbowError(msg)
 
-        if allowed_types & pygit2.credentials.GIT_CREDTYPE_USERPASS_PLAINTEXT:
+        if (allowed_types &
+                pygit2.credentials.GIT_CREDENTIAL_USERPASS_PLAINTEXT):
```
Comment (kszucs) [inline dev/archery/archery/crossbow/core.py:198]: The earlier deprecated constant has been removed in pygit 1.6: https://github.com/libgit2/pygit2/blob/master/CHANGELOG.rst#160-2021-06-01

**Comment by kszucs** [comment]
@github-actions crossbow submit *-arm64

File: dev/archery/archery/crossbow/core.py (hunk 1 above)
Comment (jonkeane) [reply dev/archery/archery/crossbow/core.py:198]: I wish the changelog had a bit more info than the declarative "deprecated GIT_CREDTYPE.*" and then "Removed deprecated GIT_CREDTYPE_*" to say _why_ without needing to go digging through git blame to find it!

**Review by jonkeane — APPROVED** [review]
This looks good, thanks

**Pull request closed without merging**